### PR TITLE
support x-ms-mutability check for read in request

### PR DESCRIPTION
**Pull request by lirenhe**
also update the gitignore file for adding test config and package-lock.json

### CODE REVIEW

File: lib/validation/custom-zschema-validators.js [hunk 1]
```diff
@@ -209,6 +209,10 @@ function isPropertyWriteOnly (xMsMutability) {
   return xMsMutability && xMsMutability.indexOf('read') === -1 && (xMsMutability.indexOf('create') != -1 || xMsMutability.indexOf('update') != -1);
 }
 
+function isPropertyReadOnly (xMsMutability) {
+  return xMsMutability && xMsMutability.indexOf('update') === -1 && (xMsMutability.indexOf('create') === -1 || xMsMutability.indexOf('read') != -1);
```
Comment (raych1) [inline lib/validation/custom-zschema-validators.js]: The last logic operation should also be `&&` coz it only allow `read` value. Can you also add comment of the expression value `"x-ms-mutability": ["read"]` for better understanding?

Comment (lirenhe) [reply lib/validation/custom-zschema-validators.js]: Updated

File: lib/validation/custom-zschema-validators.js [hunk 2]
```diff
@@ -206,7 +206,12 @@ function shouldSkipValidate (options, errors) {
 }
 
 function isPropertyWriteOnly (xMsMutability) {
-  return xMsMutability && xMsMutability.indexOf('read') === -1 && (xMsMutability.indexOf('create') != -1 || xMsMutability.indexOf('update') != -1);
+  return xMsMutability && xMsMutability.indexOf('read') === -1 && (xMsMutability.indexOf('create') != -1 || xMsMutability.indexOf('update') !== -1);
```
Comment (raych1) [inline lib/validation/custom-zschema-validators.js]: !==-1

Comment (lirenhe) [reply lib/validation/custom-zschema-validators.js]: updated

**Pull request merged**